bundles/org.openhab.binding.homewizard/src/main/java/org/openhab/binding/homewizard/internal/HomeWizardHandler.java

```diff
@@ -0,0 +1,274 @@
+            // since it is marked as required, initialize() should not be called if this field is empty
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "Missing ipAddress/host configuration");
+            return false;
+        } else {
+            updateStatus(ThingStatus.UNKNOWN);
+
+            try {
+                configurationLock.lock();
+                apiURL = String.format("http://%s/api/v1/data", config.ipAddress.trim());
+            } finally {
+                configurationLock.unlock();
+            }
+            return true;
+        }
+    }
+
+    /**
+     * Stop the poller unconditionally
+     */
+    @Override
+    public void dispose() {
+        stopPolling(false);
+    }
+
+    /**
+     * Stop the polling job
+     *
+     * @param onlyIfNeeded if true polling will only actually stop if a new refresh interval should be set
+     */
+    private void stopPolling(boolean onlyIfNeeded) {
+        try {
+            pollingJobLock.lock();
+            if (pollingJob != null && !pollingJob.isCancelled()) {
+                if (!onlyIfNeeded || config.refreshDelay != currentRefreshDelay) {
+                    if (pollingJob != null) {
+                        pollingJob.cancel(true);
+                    }
+                    pollingJob = null;
+                }
+            }
+        } finally {
+            pollingJobLock.unlock();
+        }
+    }
+
+    /**
+     * Start a polling job if it is not already running.
+     */
+    private void startPolling() {
+        try {
+            pollingJobLock.lock();
+            boolean startPoller = false;
+            if (pollingJob != null) {
+                startPoller = pollingJob.isCancelled();
+            } else {
+                startPoller = true;
+            }
+
+            if (startPoller) {
+                currentRefreshDelay = config.refreshDelay;
+                pollingJob = scheduler.scheduleWithFixedDelay(this::pollingCode, 0, currentRefreshDelay,
+                        TimeUnit.SECONDS);
+            }
+        } finally {
+            pollingJobLock.unlock();
+        }
+    }
+
+    /**
+     * The actual polling loop
+     */
+    private void pollingCode() {
+        final String query;
+        final String result;
+
+        // get a local copy of the configuration parameters
+        try {
+            configurationLock.lock();
+            query = apiURL;
+        } catch (Exception e) {
+            return;
+        } finally {
+            configurationLock.unlock();
+        }
+
+        try {
+            result = HttpUtil.executeUrl("GET", query, 750);
+        } catch (IOException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "Unable to query P1 Meter");
+            return;
+        }
+
+        if (result.trim().isEmpty()) {
+            logger.warn("P1 Wi-Fi meter API at URI {} returned empty result", apiURL);

```

Comment:
If you already update the status you should not also log to warn.
```suggestion
            logger.debug("P1 Wi-Fi meter API at URI {} returned empty result", apiURL);

```